MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -131,6 +135,30 @@ open class RouteProgress: NSObject {
         self.legIndex = legIndex
         super.init()
         currentLegProgress = RouteLegProgress(leg: currentLeg, stepIndex: 0, alertLevel: alertLevel)
+        
+        var coordinateIndex = 0
+        for leg in route.legs {
+            
+            var congestionTravelTimesSegmentsByLeg: [[(CongestionLevel, TimeInterval)]] = []
+            
+            for step in leg.steps {
+                if let lastCoord = step.coordinates?.last, let coordinates = route.coordinates, let segmentCongestionLevels = leg.segmentCongestionLevels, let expectedSegmentTravelTimes = leg.expectedSegmentTravelTimes  {
+                    guard let nextIndex = coordinates.index(where: {
+                        $0.latitude == lastCoord.latitude && $0.longitude == lastCoord.longitude
```

Comment:
Yeah, comparing doubles is bound to fail. Instead, let's take advantage of the fact that `nextIndex` is always `coordinateIndex + step.coordinates.count` to avoid a somewhat expensive call to `index(where:)`.